src/view/components/microbit/MicrobitSimulator.tsx

```diff
@@ -0,0 +1,138 @@
+    [0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0],
+];
+
+interface IState {
+    active_editors: string[];
+    running_file: string;
+    leds: number[][];
+    play_button: boolean;
+    selected_file: string;
+}
+export class MicrobitSimulator extends React.Component<any, IState> {
+    constructor() {
+        super({});
+        this.state = {
+            leds: initialLedState,
+            play_button: false,
+            selected_file: "",
+            active_editors: [],
+            running_file: "",
+        };
+    }
+    handleMessage = (event: any): void => {
+        const message = event.data;
+
+        switch (message.command) {
+            case "reset-state":
+                this.setState({
+                    leds: initialLedState,
+                    play_button: false,
+                });
+                break;
+            case "set-state":
+                this.setState({
+                    leds: message.state.leds,
+                });
+                break;
+            case "activate-play":
+                this.setState({
+                    play_button: !this.state.play_button,
+                });
+                break;
+            case "visible-editors":
+                this.setState({
+                    active_editors: message.state.activePythonEditors,
+                });
+                break;
+            case "current-file":
+                this.setState({
+                    running_file: message.state.running_file,
+                });
+                break;
+            default:
+                console.log("Invalid message received from the extension.");
+                break;
+        }
+    };
+    componentDidMount() {
+        window.addEventListener("message", this.handleMessage);
+    }
+    componentWillUnmount() {
+        window.removeEventListener("message", this.handleMessage);
+    }
+
+    render() {
+        const playStopImage = this.state.play_button ? StopLogo : PlayLogo;
+
+        return (
+            <div className="simulator">
+                <div className="file-selector">
+                    <Dropdown
+                        label={"file-dropdown"}
+                        styleLabel={"dropdown"}
+                        lastChosen={this.state.running_file}
+                        width={300}
+                        textOptions={this.state.active_editors}
+                        onBlur={this.onSelectFile}
+                    />
+                </div>
+                <div className="microbit-container">
+                    <MicrobitImage
+                        eventTriggers={{
+                            onMouseDown: this.onMouseDown,
+                            onMouseUp: this.onMouseUp,
+                            onMouseLeave: this.onMouseLeave,
+                        }}
+                        leds={this.state.leds}
+                    />
+                </div>
+                <ActionBar

```

Comment:
This Action bar is separate from the one at the root level right?

Reply:
the root level is the tab, the action bar is toggle play

Reply:
sounds good